db/fixtures/ae_datastore/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb

```diff
@@ -0,0 +1,188 @@
+  when :storage
+    requested_storage(prov_option, number_of_vms, options_array)
+  else
+    options_value(prov_option, @miq_request, options_array)
+  end
+  options_array
+end
+
+def requested_number_of_cpus(number_of_vms, options_array)
+  cpu_in_request = get_option_value(@miq_request, :number_of_cpus)
+  if cpu_in_request.zero?
+    cpu_in_request = get_option_value(@miq_request, :number_of_sockets) *
+                     get_option_value(@miq_request, :cores_per_socket)
+  end
+  total_cpu = number_of_vms * cpu_in_request
+  set_requested_value(prov_option, total_cpu,
+                      @miq_request.get_option(:instance_type), options_array)
+end
+
+def requested_storage(prov_option, number_of_vms, options_array)
+  vm_size = @miq_request.vm_template.provisioned_storage  / 1024**2
+  total_storage = number_of_vms * vm_size
+  set_requested_value(prov_option, total_storage,
+                      @miq_request.get_option(:instance_type), options_array)
+end
+
+def options_value(prov_option, resource, options_array)
+  return unless resource.respond_to?('get_option')
+  set_requested_value(prov_option, resource.get_option(prov_option),
+                      resource.get_option(:instance_type), options_array)
+end
+
+def collect_totals(array)
+  array.collect(&:to_i).inject(&:+).to_i
+end
+
+def collect_dialog_totals(prov_option, options_hash)
+  dialog_values(prov_option, options_hash, dialog_array = [])
+  collect_totals(dialog_array)
+end
+
+def request_totals(template_totals, dialog_totals)
+  template_totals < dialog_totals ? dialog_totals : template_totals
+end
+
+def dialog_values(prov_option, options_hash, dialog_array)
+  options_hash.each do |_sequence_id, options|
+    set_requested_value(prov_option, options[prov_option], options[:instance_type], dialog_array)
+  end
+end
+
+def set_requested_value(prov_option, option_value, find_id, dialog_array)
+  $evm.log(:info, "set requested value: prov_option: #{prov_option} value: #{option_value}")
+  return if cloud_value(prov_option, option_value, find_id, dialog_array)
+
+  $evm.log(:info, "set requested value default_option: prov_option: #{prov_option} value: #{option_value}")
+  default_option(option_value, dialog_array)
+end
+
+def cloud_value(prov_option, option_value, find_id, dialog_array)
+  return nil unless @cloud
+
+  case prov_option
+  when :cores_per_socket
+    $evm.log(:info, "set requested value cores_per_socket: prov_option: #{prov_option} value: #{option_value}")
+    option_set = requested_cores_per_socket(find_id, dialog_array)
+  when :vm_memory
+    $evm.log(:info, "set requested value vm_memory: prov_option: #{prov_option} value: #{option_value}")
+    option_set = requested_vm_memory(find_id, dialog_array)
+  when :storage
+    $evm.log(:info, "set requested value storage: prov_option: #{prov_option} value: #{option_value}")
+    option_set = requested_storage(@miq_request.get_option(:src_vm_id), dialog_array)
+  end
+  option_set
```

Comment:
@tinaafitz 
If its not :cores_per_socket, :vm_memory, :storage should we log and is it ok for the option_set to be nil since it won't be set.